lib/interactive_viewer.ml

```diff
@@ -149,9 +156,19 @@ let view (patches : Patch.t list) =
           operation_info z_patches;
           change_summary z_patches;
           current_operation z_patches;
-          W.vscroll_area
-            ~state:(Lwd.get curr_scroll_state)
-            ~change:change_scroll_state
+          (* W.vscroll_area
+               ~state:(Lwd.get curr_scroll_state)
+               ~change:change_scroll_state
+             @@ W.hscroll_area
+                  ~state:(Lwd.get curr_scroll_state_h)
+                  ~change:change_scroll_state_h
+             @@ current_hunks z_patches; *)
+          W.hscroll_area
+            ~state:(Lwd.get curr_scroll_state_h)
+            ~change:change_scroll_state_h
+          @@ W.vscroll_area
+               ~state:(Lwd.get curr_scroll_state)
+               ~change:change_scroll_state
           @@ current_hunks z_patches;
           Lwd.pure
           @@ Ui.keyboard_area
```

Comment:
> When using a vscroll area inside hscroll area, I've noticed that the horizontal scrolling behaves correctly with the left and right keys. However, there's an issue with vertical scrolling – when using the up and down keys, the behavior is unexpected, and vice versa.

Can you describe the unexpected behavior and how to get to it ?

I notice one unexpected behavior:
On a hunk that is big enough to scroll both vertically and horizontally, the right key has no effect when pressed but when pressing the down key, update both the vertical and horizontal scroll to the expected position.

I'm not sure why this happens by reading the code. Could you try printing the values of the states. Here's a setup that can work with two terminals: In one terminal run the app with `dune exec -- diffcessible example.diff 2> stderr`, in the second terminal read this file with `tail -f stderr`. You can then interact with the app and see debug printing at the same time. Make sure to print to stderr, for example with `Printf.eprintf`.
Good luck!

Reply:
> Can you describe the unexpected behavior and how to get to it ?
> 
> I notice one unexpected behavior: On a hunk that is big enough to scroll both vertically and horizontally, the right key has no effect when pressed but when pressing the down key, update both the vertical and horizontal scroll to the expected position.
> 
> I'm not sure why this happens by reading the code. Could you try printing the values of the states. Here's a setup that can work with two terminals: In one terminal run the app with `dune exec -- diffcessible example.diff 2> stderr`, in the second terminal read this file with `tail -f stderr`. You can then interact with the app and see debug printing at the same time. Make sure to print to stderr, for example with `Printf.eprintf`. Good luck!

I'll investigate further and try to understand why the right key is not having any effect, I'll set up the suggested environment with two terminals as you described.
Once I have more information from the debug prints, I'll analyze the behavior and work on resolving the issue accordingly.

Thanks for your guidance, and I'll keep you updated on my progress!